src/main/js/field/EndpointSelectField.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createNewConfigurationRequest } from 'util/configurationRequestBuilder';
+import DynamicSelectInput from 'field/input/DynamicSelect';
+import * as FieldModelUtilities from 'util/fieldModelUtilities';
+
+class EndpointSelectField extends Component {
+    constructor(props) {
+        super(props);
+
+        this.onSendClick = this.onSendClick.bind(this);
+
+        this.state = ({
+            options: [],
+            progress: false,
+        });
+    }
+
+    componentDidUpdate(prevProps) {
+        const oldValuesEmpty = FieldModelUtilities.areKeyToValuesEmpty(prevProps.currentConfig);
+        const currentValuesEmpty = FieldModelUtilities.areKeyToValuesEmpty(this.props.currentConfig);
+        if (oldValuesEmpty && !currentValuesEmpty) {
+            this.onSendClick();
+        }
+    }
+
+    onSendClick() {
+        this.setState({
+            fieldError: this.props.errorValue,
+            success: false
+        });
+        const {
+            fieldKey, csrfToken, currentConfig, endpoint
+        } = this.props;
+
+        const request = createNewConfigurationRequest(`/alert${endpoint}/${fieldKey}`, csrfToken, currentConfig);
+        request.then((response) => {
+            if (response.ok) {
+                response.json().then((data) => {
+                    const options = data.map(item => {
+                        const dataValue = item.value;
+                        return { icon: null, key: dataValue, label: dataValue, value: dataValue };
+                    });
+
+                    this.setState({
+                        options,
+                        success: true
+                    });
+                });
+
+            } else {
+                response.json().then((data) => {
+                    this.setState({
+                        options: [],
+                        fieldError: data.message
+                    });
+                });
+            }
+        });
+    }
+
+    render() {
+        return (
+            <div>
+                <DynamicSelectInput onChange={this.props.onChange} onFocus={this.onSendClick} options={this.state.options} {...this.props} />
```

Comment:
I still think we should be using AsyncSelect here.

Reply:
Just looked more into AsyncSelect, the only real benefit would be async functionality. I don't really see any other benefits from it. If we use the AsyncSelect, we'll need to add more code for styling, state changes, content and other things which is already done in DynamicSelectInput.

Reply:
The benefit of the AsyncSelect is that it will handle loading the data by submitting a request.  It will handle displaying that data is being loaded.  It's a component that has been built for this purpose.  DynamicSelectInput is our own home grown input field for select inputs that we are now retro-fitting to be an asynchronous select field.

Reply:
The only real change we made to DynamicSelectInput was to allow someone to pass the underlying Select an additional property. For the AsyncSelect, we still had to make the request and pass it through. The displaying is now being handled by our own Select. So it still seems to me like we're only missing out on async functionality here.